tests/api.rs

```diff
@@ -0,0 +1,37 @@
+mod support;
+
+#[cfg(feature = "api")]
+mod tests {
+    use crate::support::{sink, source};
+    use vector::config::Config;
+    use vector::test_util::{next_addr, start_topology, wait_for_tcp_in_secs};
+
+    fn api_enabled_config() -> Config {
+        let mut config = Config::empty();
+        config.add_source("in1", source().1);
+        config.add_sink("out1", &["in1"], sink(10).1);
+        config.api.enabled = true;
+        config.api.bind = Some(next_addr());
+
+        config
+    }
+
+    #[tokio::test]
+    async fn api_config() {
+        let config = api_enabled_config();
+        let addr = config.api.bind.unwrap();
+
+        let (mut _top, _) = start_topology(config, false).await;
+        wait_for_tcp_in_secs(addr, 30).await;
```

Comment:
It is great to have the test, though I'm not a big fan of timers.
It would be cool if the topology exposed a channel for when the API setup is complete and the socket is listening - we could eliminate the race here, and potentially also utilize this for health checks.

Reply:
Yeah, I just reached for this since it was already in use.

Reply:
I think we're generally moving away from using timers in tests where possible. They caused us some unncessary maintenance effort in the past.

Reply:
Happy to switch this to something else, but didn't want to touch topology too much in this first PR, in case it impacted other areas I'm not very familiar with.